gcc/rust/backend/rust-mangle.cc

```diff
@@ -279,26 +314,155 @@ v0_add_identifier (std::string &mangled, const std::string &identifier)
   std::replace (punycode.begin (), punycode.end (), '-', '_');
 
   if (!is_ascii_ident)
-    mangled.append ("u");
+    mangled << "u";
 
-  mangled += std::to_string (punycode.size ());
+  mangled << std::to_string (punycode.size ());
   // If the first character of the identifier is a digit or an underscore, we
   // add an extra underscore
   if (punycode[0] == '_')
-    mangled += "_";
+    mangled << "_";
 
-  mangled += punycode;
+  mangled << punycode;
+  return mangled.str ();
 }
 
 static std::string
-v0_type_prefix (const TyTy::BaseType *ty)
+v0_path (const TyTy::BaseType *ty, const Resolver::CanonicalPath &path)
 {
-  auto ty_prefix = v0_simple_type_prefix (ty);
-  if (!ty_prefix.empty ())
-    return ty_prefix;
+  auto mappings = Analysis::Mappings::get ();
 
-  // FIXME: We need to fetch more type prefixes
-  rust_unreachable ();
+  std::string mangled;
+
+  path.iterate_segs ([&] (const Resolver::CanonicalPath &seg) {
```

Comment:
I think this lambda is very long and would benefit from being split up in multiple tinier functions - but this can be addressed later  before the PR gets merged

Reply:
Thanks. Fixed in this PR.